### DISCUSSION TITLE
Store feature variant for each user in database

### DISCUSSION THREAD

**Discussion by energetiks** [Unleash/unleash#7204] [Q&A] [answered]
Hello! I have a question about storing information about feature variants in database
So, what i did
1. I have running unleash instance. It works perfectly
2. I create new feature toggle called "tymy", create four variants
3. I connect to my unleash instance through python and get variant 

`from UnleashClient import UnleashClient`

`client = UnleashClient(`
 `   url="http://localhost:4242/api",`
`    app_name="default",`
`    custom_headers={'Authorization': '*'})`
`client.initialize_client()`
`context = {'userId': 'testtymy1'}`
`variant = client.get_variant("tymy", context)`
`print(variant)`

4. Everything works perfectly at this step! For user with id "testtymy1" I got variant "test1"

And finally i need to find in unleash postgres database information that user with id "testtymy" has variant "test1". I need this for using this information in my analytic systems
But unfortunately i couldn't  find this info in database and there are no mentions about it in documentation

Could you help me please?

**Comment by chriswk** [comment, marked as answer]
Hi. I think there's a misunderstanding here with regards to how Unleash evaluates toggles/variants. 

There will be no trace in the database that user with id testtymy has variant test1.

We only store the rules for configuring the SDK in the database.

You're probably looking for impression data: https://docs.getunleash.io/reference/impression-data
